### ISSUE TITLE
Add code formatting

### ISSUE THREAD

**Issue by Igor-Mochalin** [jshinonome/vscode-q#18]
A lot of devs write q code as one unformatted line.. sometimes it's quite hard to parse)).. will be good to have code formatting as in other popular languages like java,c,....

**Comment by jshinonome** [comment]
Yes, of course. but as you know, how unpredictable about q code. Will revisit this if I got time to do this.

**Comment by BenjaminLiuPenrose** [comment]
Hello, out of curiosity, do you know a good q code formatting standard?

**Comment by jshinonome** [comment]
Unfortunately, I don't have any good format standard yet. I would need some advice on this later. For now I will create something based on my experience.

**Comment by codereport** [comment]
I am unfamiliar with Tree-Sitter but would love to help materialize a formatter for q. It doesn't seem that [Tree-sitter](https://tree-sitter.github.io/tree-sitter/) itself has support for code formatting but there is a recently new project called [Topiary](https://github.com/tweag/topiary) (this is an [announcement blog](https://www.tweag.io/blog/2023-03-09-announcing-topiary/) as well) that leverages Tree-sitter to perform formatting. I will poke around to see if I can get a simple proof of concept working.

**Comment by jshinonome** [comment]
It is implemented on vscode-k-pro. No plan for vscode-q.
